plugins/module_utils/network/sonic/config/dcbx/dcbx.py

```diff
@@ -0,0 +1,440 @@
+        return commands, requests
+
+    def _state_merged(self, diff, have):
+        """ The command generator when state is merged
+        :rtype: A list
+        :returns: the commands necessary to merge the provided into
+                  the current configuration
+        """
+        commands = diff
+        requests = []
+        requests = self.get_modify_specific_dcbx_interfaces_param_requests(
+            commands)
+
+        if commands and len(requests) > 0:
+            commands = update_states(commands, 'merged')
+        else:
+            commands = []
+        return commands, requests
+
+    def get_length(self, obj):
+        if isinstance(obj, list):
+            if obj and isinstance(obj[0], dict):
+                return len(obj[0])
+            else:
+                return 0
+        elif isinstance(obj, dict):
+            return len(obj)
+        else:
+            return 0
+
+    def remove_default_entries(self, data):
+        pop_list = []
+        new_data = {}
+
+        glbl = data.get('global')
+        if glbl:
+            enabled = glbl.get('enabled')
+            if enabled is not True:
+                glbl.pop('enabled')
+
+        intfs = data.get('interfaces')
+        if intfs:
+            for intf in intfs:
+                pop_item = False
+                if 'pfc_tlv_enabled' in intf and intf.get('pfc_tlv_enabled') is True:
+                    intf.pop('pfc_tlv_enabled')
+                    pop_item = True
+                if 'ets_configuration_tlv_enabled' in intf and intf.get('ets_configuration_tlv_enabled') is True:
+                    intf.pop('ets_configuration_tlv_enabled')
+                    pop_item = True
+                if 'ets_recommendation_tlv_enabled' in intf and intf.get('ets_recommendation_tlv_enabled') is True:
+                    intf.pop('ets_recommendation_tlv_enabled')
+                    pop_item = True
+                if 'enabled' in intf and intf.get('enabled') is True:
+                    intf.pop('enabled')
+                    pop_item = True
+                if 'name' in intf and len(intf) == 1 and pop_item:
+                    idx = intfs.index(intf)
+                    pop_list.insert(0, idx)
+
+            for idx in pop_list:
+                intfs.pop(idx)
+            if not intfs:
+                data.pop('interfaces')
+
+    def _state_deleted(self, want, have, diff):
+        """ The command generator when state is deleted
+        :rtype: A list
+        :returns: the commands necessary to remove the current configuration
+                  of the provided objects
+        """
+        commands = []
+        requests = []
+
+        is_delete_all = False
+        if not want:
+            commands = deepcopy(have)
+            is_delete_all = True
+        else:
+            commands = get_diff(want, diff, TEST_KEYS)
+        self.remove_default_entries(commands)
+        commands = remove_empties(commands)
```

Comment:
Please see the related comments above regarding the usage of "remove_empties" in this file.

For this case, the only thing needed is to remove the empties from any specified interfaces in the input commands. The following suggestion would accomplish this:

```suggestion
        intf_commands = commands.get('interfaces')
        if intf_commands:
            commands['interfaces'] = remove_empties_from_list(intf_commands)
```

Reply:
Done.